s3-vfs/src/main/java/org/pentaho/s3common/S3CommonFileObject.java

```diff
@@ -411,6 +413,23 @@ protected void doRename( FileObject newFile ) throws Exception {
     delete();
   }
 
+  private void doFolderMove( FileObject sourceFolder, FileObject targetFolder ) throws FileSystemException {
```

Comment:
Would love to have automation actually testing this. I wonder how we could do that. Are there any steps or job entries that move folders?

Reply:
There is a move entry that could work with vfs. And we can always code some integration tests